distributed/protocol/serialize.py

```diff
@@ -71,14 +74,55 @@ def pickle_loads(header, frames):
     return pickle.loads(x, buffers=buffers)
 
 
+def import_allowed_module(name):
+    if name in _cached_allowed_modules:
+        return _cached_allowed_modules[name]
+
+    if name.split(".")[0] in dask.config.get("distributed.scheduler.allowed-imports"):
```

Comment:
@madsbk I relaxed the check here a bit to just check the root module being imported. So, for example, `dask.highlevelgraph`, `dask.dataframe.io.parquet.core`, etc. will all match against "dask" in the list of allowed imports. My thought is that this will make things more flexible, while still providing a reasonable safeguard against importing arbitrary modules. Curious to hear what you think

Reply:
I think you are right. Python doesn't allow mixing relative and absolute import paths so it should be boxed in. I have added some extra checks to be on the safe side.